diff -rupN ./src/Eigen/unsupported/Eigen/src/NumericalDiff/NumericalDiff.h ../external/src/Eigen/unsupported/Eigen/src/NumericalDiff/NumericalDiff.h
--- ./src/Eigen/unsupported/Eigen/src/NumericalDiff/NumericalDiff.h	2015-06-25 16:23:41.000000000 +0200
+++ ../external/src/Eigen/unsupported/Eigen/src/NumericalDiff/NumericalDiff.h	2015-01-07 10:58:02.000000000 +0100
@@ -52,6 +52,10 @@ public:
         NumericalDiff(const T0& a0, const T1& a1) : Functor(a0, a1), epsfcn(0) {}
     template<typename T0, typename T1, typename T2>
         NumericalDiff(const T0& a0, const T1& a1, const T2& a2) : Functor(a0, a1, a2), epsfcn(0) {}
+    template<typename T0, typename T1, typename T2, typename T3>
+        NumericalDiff(const T0& a0, const T1& a1, const T2& a2, const T3& a3) : Functor(a0, a1, a2, a3), epsfcn(0) {}
+    template<typename T0, typename T1, typename T2, typename T3, typename T4>
+        NumericalDiff(const T0& a0, const T1& a1, const T2& a2, const T3& a3, const T4& a4) : Functor(a0, a1, a2, a3, a4), epsfcn(0) {}
 
     enum {
         InputsAtCompileTime = Functor::InputsAtCompileTime,
