--- ./src/libjpeg-turbo/simd/jsimd_arm64_neon.S	2016-03-02 16:04:36.000000000 +0000
+++ ./src/libjpeg-turbo/simd/jsimd_arm64_neon_patched.S	2016-03-16 14:09:41.000000000 +0000
@@ -30,7 +30,13 @@
 #endif
 
 .text
-.arch armv8-a+fp+simd
+//  16 Mar 2016, Steve Madsen, Blippar, London UK
+//  Commented out due to build error w/clang in latest NDK bundle
+//      jni/../../../../External/android/../src/libjpeg-turbo/simd/jsimd_arm64_neon.S:33:1: error: unknown directive
+//      .arch armv8-a+fp+simd
+//  Refs
+//      https://github.com/libjpeg-turbo/libjpeg-turbo/issues/33
+//.arch armv8-a+fp+simd
 
 
 #define RESPECT_STRICT_ALIGNMENT 1
